### PR TITLE
Add laser tower

### LINKED ISSUE #251: Add laser tower
- no cooldown
- increase damage over time when target is linked
- range?
- change target when target move out of range

### PR DESCRIPTION
**Pull request by tanwayne890**
Closes #251 

This PR adds:
- upgradeable and elemental laser towers (models given by @Eriksen2411 )
- lasers effect (line renderer, particles, glow and light), color according to its elements (default is green)
- laser tower icons into each loadout 

Screenshots (Videos in tele):

image

image

image

image
(slash-ed means not used)

### CODE REVIEW

File: Assets/Scripts/Towers/LaserTower.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+        // Enemy target lock on 
+        Vector3 dir = target.position - transform.position;
+        Quaternion lookAtRotation = Quaternion.LookRotation(dir);
+        Vector3 rotation = Quaternion.Lerp(rotationBase.rotation, lookAtRotation, Time.deltaTime * rotationSpeed).eulerAngles;
+        rotationBase.rotation = Quaternion.Euler(0f, rotation.y, 0f);
+
+        // laser damage
+        targetTime += Time.deltaTime;
+        targetEnemy.TakeDamage(damage * Mathf.Log(targetTime + 1), elementInfo); // +1 so that is positive
```
Comment (tanwayne890) [inline Assets/Scripts/Towers/LaserTower.cs:88]: damage * log(accumulatedTime + 1). +1 is to start from 0

File: Assets/Scripts/Enemy/Enemy.cs [hunk 2]
```diff
@@ -127,8 +127,8 @@ public class Enemy : MonoBehaviour {
             // update health bar, float number between 0 and 1
             healthBar.fillAmount = health / enemyInfo.health;
 
-            DamageIndicator indicator = Instantiate(damageText, transform.position, Quaternion.identity).GetComponent<DamageIndicator>();
-            indicator.SetDamageTextFromFloat(temp);
+            //DamageIndicator indicator = Instantiate(damageText, transform.position, Quaternion.identity).GetComponent<DamageIndicator>();
+            //indicator.SetDamageTextFromFloat(temp);
```
Comment (tanwayne890) [inline Assets/Scripts/Enemy/Enemy.cs:131]: removed damage number on enemy since it looks ugly

File: Assets/Scripts/Enemy/Enemy.cs [hunk 3]
```diff
@@ -224,7 +224,7 @@ public class Enemy : MonoBehaviour {
     }
 
     private void Update() {
-        healthText.text = string.Format("{0}", health);
+        healthText.text = string.Format("{0:N0}", health);
```
Comment (tanwayne890) [inline Assets/Scripts/Enemy/Enemy.cs:227]: set health text to no decimal

**Comment by Timothyoung97** [comment]
Laser Tower tested:
- Tower works as expected.
- Upgrade Mechanics works as well.

Game Feel:
- I feel that the tower is too strong for now, but we can do an overall game balance in the future.

Code:
- lgtm

**Pull request merged**